.github/workflows/tag_and_create_release.yaml

```diff
@@ -0,0 +1,59 @@
+name: Tag and Create Github Release
+
+on:
+  push:
+    branches: [main]
+    paths: [CHANGELOG.md, pubspec.yaml]
```

Comment:
Note: This workflow will trigger if _either_ of these files change. For additional "safety", we check the name of the `pusher` (which should be the bot).